src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from "react";
+import { Todo } from "../../types";
+import usersFromServer from '../../api/users';
+import { getUserById } from '../../App';
+
+type Props = {
+  onAdd: (todo: Todo) => void,
+  todos: Todo[],
+};
+
+
+const getNewTodoId = (todos: Todo[]): number => {
+  return todos.reduce((maxId, todo: Todo) => {
+    if (todo.id > maxId) {
+      return todo.id;
+    }
+
+    return maxId;
+  }, 0) + 1;
+};
+
+export const TodoForm: React.FC<Props> = ({
+  onAdd,
+  todos,
+}) => {
+  const [title, setTitle] = useState('');
+  const [isTitleFailed, setIsTitleFailed] = useState(false);
+
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [isSelectedUserIdFailed, setSelectedUserIdFailed] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setIsTitleFailed(true);
+    };
+
+    if (!selectedUserId) {
+      setSelectedUserIdFailed(true);
+    };
+
+    if (!title || !selectedUserId) {
+      return;
+    };
+
+    const todo: Todo = {
+      title,
+      user: getUserById(selectedUserId),
+      id: getNewTodoId(todos),
+      completed: false,
+      userId: selectedUserId,
+    };
+
+    onAdd(todo);
+  };
+
+  const handleTitleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
```

Comment:
Something wrong with the spacing
"Expected indentation of 2 spaces but found 4"